src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
-            negative_type,
-            ..Default::default()
-        };
-        test_template::<F, PC>(info)
-    }
+            // (X - x_i)
+            let x_polynomial =
+                Polynomial::from_coefficients_slice(&[-(*point), G::ScalarField::one()]);
 
-    pub fn single_poly_degree_bound_multiple_queries_test<F, PC>(
-        negative_type: Option<NegativeType>,
-    ) -> Result<(), PC::Error>
-    where
-        F: Field,
-        PC: PolynomialCommitment<F>,
-    {
-        let info = TestInfo {
-            num_iters: 100,
-            max_degree: None,
-            supported_degree: None,
-            num_polynomials: 1,
-            enforce_degree_bounds: true,
-            max_num_queries: 2,
-            negative_type,
-            ..Default::default()
-        };
-        test_template::<F, PC>(info)
-    }
+            // z_i(x)/z(x) = 1 / (x - x_i).
+            // unwrap cannot fail as x-x_i is guaranteed to be non-zero.
+            let z_i_over_z_value = x_polynomial.evaluate(x_point).inverse().unwrap();
 
-    pub fn two_polys_degree_bound_single_query_test<F, PC>(
-        negative_type: Option<NegativeType>,
-    ) -> Result<(), PC::Error>
-    where
-        F: Field,
-        PC: PolynomialCommitment<F>,
-    {
-        let info = TestInfo {
-            num_iters: 100,
-            max_degree: None,
-            supported_degree: None,
-            num_polynomials: 2,
-            enforce_degree_bounds: true,
-            max_num_queries: 1,
-            negative_type,
-            ..Default::default()
-        };
-        test_template::<F, PC>(info)
-    }
+            lc_commitment +=
+                &(labeled_commitment.commitment().clone() * &(z_i_over_z_value * cur_challenge));
 
-    pub fn full_end_to_end_test<F, PC>(negative_type: Option<NegativeType>) -> Result<(), PC::Error>
-    where
-        F: Field,
-        PC: PolynomialCommitment<F>,
-    {
-        let info = TestInfo {
-            num_iters: 100,
-            max_degree: None,
-            supported_degree: None,
-            num_polynomials: 10,
-            enforce_degree_bounds: true,
-            max_num_queries: 5,
-            negative_type,
-            ..Default::default()
-        };
-        test_template::<F, PC>(info)
-    }
+            lc_value += y_i * cur_challenge * z_i_over_z_value;
 
-    pub fn segmented_test<F, PC>(negative_type: Option<NegativeType>) -> Result<(), PC::Error>
-    where
-        F: Field,
-        PC: PolynomialCommitment<F>,
-    {
-        let info = TestInfo {
-            num_iters: 100,
-            max_degree: None,
-            supported_degree: None,
-            num_polynomials: 10,
-            enforce_degree_bounds: true,
-            max_num_queries: 5,
-            segmented: true,
-            negative_type,
-            ..Default::default()
-        };
-        test_template::<F, PC>(info)
-    }
+            // lambda^i
+            cur_challenge = cur_challenge * &lambda;
+        }
```

Comment:
Also here, just for completness: One could compute the coefficients of the LC alone, and call the lc combine afterwards.